feature/metis/conversation/src/main/kotlin/de/tum/informatics/www1/artemis/native_app/feature/metis/conversation/ui/ConversationViewModel.kt

```diff
@@ -165,6 +166,21 @@ internal open class ConversationViewModel(
     }
         .stateIn(viewModelScope + coroutineContext, SharingStarted.Eagerly, false)
 
+    val isAtLeastTutorInCourse: StateFlow<Boolean> = flatMapLatest(
+        serverConfigurationService.serverUrl,
+        accountService.authToken,
+        onRequestReload.onStart { emit(Unit) }
+    ) { serverUrl, authToken, _ ->
+        retryOnInternet(networkStatusProvider.currentNetworkStatus) {
+            accountDataService.getAccountData(
+                serverUrl = serverUrl,
+                bearerToken = authToken
+            )
+                .bind { it.isAtLeastTutorInCourse(course = this.course.value.orThrow()) }
```

Comment:
Why do you go for course.value? If the course flow emits a new value, then this flow should also be reexecuted. Therefore, the course flow should be a parameter in flatmaplatest